runtime/planner.js

```diff
@@ -132,62 +132,114 @@ class Planner {
     return result.some(r => r.score == 0);
   }
 
+  _speculativeThreadCount() {
+    // TODO(wkorman): We'll obviously have to rework the below when we do
+    // speculation in the cloud.
+
+    // Logical processor cores.
+    const hasNavigator = typeof navigator !== 'undefined';
```

Comment:
This should be a platform/ abstraction. runtime/ shouldn't include browser specific logic.

Reply:
OK, it looks like the current examples are structured basically relying on this:

https://github.com/PolymerLabs/arcs/blob/master/tools/custom-loader.mjs#L18

So I'll move accessing the two `navigator` methods to something like `deviceinfo-web.js` and have a `deviceinfo-node.js` that just returns zero for these methods, or let me know if other thoughts.

There is one different pattern used by `devtools-channel` which has a provider, but presumably that's due to extension specific antics needed.